tests/waku_rln_relay/test_waku_rln_relay.nim

```diff
@@ -632,30 +632,25 @@ suite "Waku rln relay":
 
     # check whether hasDuplicate correctly finds records with the same nullifiers but different secret shares
     # no duplicate for proof1 should be found, since the log is empty
```

Comment:
if 

> This could lead to a node only accepting one valid message sent during an epoch e, and reject all other messages sent during epoch e.

would it be difficult to reproduce it with a test? would like to i) verify that its indeed the case and ii) protect the code from future modification on such critical thing.

Reply:
addressed in 6d20aef7
sorry, that statement is a little unclear. This bug is mostly to do with the nullifierLog not being mapped from Epoch to ProofMetadata, but rather ExternalNullifier to ProofMetadata. this line https://github.com/waku-org/nwaku/blob/88ff928213b92d20b54453687823afa510ee8d1a/waku/waku_rln_relay/rln_relay.nim#L117 does in fact allow the node to accept multiple valid messages sent during an epoch `e` from multiple senders.

Reply:
because the proofMetadata will be different from different senders. count this PR as a fix for the Table, and optimization to identify double signaling